### PR TITLE
Feature/sergeyro/css modules migration 8

### PR DESCRIPTION
**Pull request by SergeyRoyt**
- [ ] 1. Banner/Banner.jsx
- [ ] 2. BreadcrumbsBar/BreadcrumbItem/BreadcrumbContent/BreadcrumbContent.jsx
- [ ] 3. BreadcrumbsBar/BreadcrumbItem/BreadcrumbItem.jsx
- [ ] 4. BreadcrumbsBar/BreadcrumbsBar.jsx
- [ ] 5. Button/Button.jsx
- [ ] 6. ButtonGroup/ButtonGroup.jsx

### CODE REVIEW

File: src/components/Banner/Banner.jsx [hunk 1]
```diff
@@ -34,26 +38,26 @@ const Banner = forwardRef(
     const renderedTitle = useMemo(() => {
       const computedTitle = renderTitle(title);
       if (!computedTitle) return null;
-      return <h2 className="banner--title">{computedTitle}</h2>;
+      return <h2 className={cx(styles.bannerTitle, "banner--title")}>{computedTitle}</h2>;
```
Comment (hadasfa) [inline src/components/Banner/Banner.jsx:41]: didn't we decide to remove old classnames?

File: src/components/Banner/Banner.jsx [hunk 2]
```diff
@@ -65,11 +69,27 @@ const Banner = forwardRef(
...
         <div
-          className={cx("banner--content", `image-position__${imagePosition}`, {
-            "close-button-spacing": !!renderCloseButton
-          })}
+          id={id}
+          data-testid={dataTestId || getTestId(ELEMENT_TYPES.BANNER, id)}
+          className={cx(
+            styles.bannerContent,
+            "banner--content",
+            styles[`${camelCase("image-position__" + imagePosition)}`],
```
Comment (hadasfa) [inline src/components/Banner/Banner.jsx:86]: why not just [`imagePosition${imagePosition}`]

Comment (hadasfa) [reply src/components/Banner/Banner.jsx:86]: it's just confusing this way

File: src/components/Button/Button.jsx [hunk 3]
```diff
@@ -132,22 +133,37 @@ const Button = forwardRef(
       const calculatedColor = success ? BUTTON_COLORS.POSITIVE : color;
       return cx(
         className,
+        styles.button,
         "monday-style-button",
+        styles[`${camelCase("size-" + getActualSize(size))}`],
```
Comment (hadasfa) [inline src/components/Button/Button.jsx:138]: Why not use more readable syntax here? without the camel case function.

File: src/components/Button/Button.jsx [hunk 4]
```diff
@@ -132,22 +133,37 @@ const Button = forwardRef(
...
+        styles[`${camelCase("color-" + calculatedColor)}`],
         `monday-style-button--color-${calculatedColor}`,
         {
-          "has-style-size": hasSizeStyle,
-          "monday-style-button--loading": loading,
+          [styles.hasStyleSize]: hasSizeStyle,
+          ["has-style-size"]: hasSizeStyle,
+          [styles.loading]: loading,
+          ["monday-style-button--loading"]: loading,
+          [`${camelCase("color-" + calculatedColor + "-active")}`]: active,
```
Comment (hadasfa) [inline src/components/Button/Button.jsx:149]: same camel case function note

File: src/components/ButtonGroup/ButtonGroup.jsx [hunk 5]
```diff
@@ -76,10 +83,13 @@ const ButtonGroup = forwardRef(
             tooltipShowDelay={tooltipShowDelay}
             tooltipContainerSelector={tooltipContainerSelector}
             tooltipMoveBy={tooltipMoveBy}
-            className={cx(`${baseClassName}__option-text`, {
-              selected: isSelected,
-              disabled,
-              "button-disabled": option.disabled
+            className={cx(styles[`${camelCase("group-component__option-text")}`], `${CSS_BASE_CLASS}__option-text`, {
```
Comment (hadasfa) [inline src/components/ButtonGroup/ButtonGroup.jsx:86]: camel case function comment

File: src/components/ButtonGroup/ButtonGroup.jsx [hunk 6]
```diff
@@ -109,19 +121,39 @@ const ButtonGroup = forwardRef(
 
     return (
       <div
-        className={cx(baseClassName, overrideClassName, `${baseClassName}--kind-${kind}`, { disabled })}
+        className={cx(
+          styles.groupComponent,
+          CSS_BASE_CLASS,
+          overrideClassName,
+          styles[`${camelCase("group-component--kind-" + kind)}`],
```
Comment (hadasfa) [inline src/components/ButtonGroup/ButtonGroup.jsx:128]: camel case function note

**Review by hadasfa — APPROVED** [review]
Hey i am approving for not being a blocker. But please remove this weird camelCase function calls for converting old classnames to css modules syntax. is confusing and add complexity and people which contribute vibe may take this as an example (we do not want that)/
overall amazing work of course!

File: src/components/Banner/Banner.jsx (hunk 2 above)
Comment (SergeyRoyt) [reply src/components/Banner/Banner.jsx:86]: imagePosition need to be converted to camelCase anyway, cause even if it's one word, it's starting with lowercase letter

Comment (orrgottlieb) [reply src/components/Banner/Banner.jsx:86]: lets leave it for now, and discuss this in the merged branch before version 2.0

**Comment by SergeyRoyt** [comment]
TODO in `Button.tsx` - styles seems so not load - seen in snapshots - problems with ts + css modules

**Comment by SergeyRoyt** [comment]
Merged into [Vibe2.0](https://github.com/mondaycom/monday-ui-react-core/pull/885)

**Pull request closed without merging**